include/pagmo/problem.hpp

```diff
@@ -564,6 +606,26 @@ inline void check_problem_bounds(const std::pair<vector_double, vector_double> &
                             + " while the upper bound has the smaller value " + std::to_string(ub[i]));
         }
     }
+    // 3 - checks the integer part
+    if (nix) {
+        auto nx = lb.size();
+        auto ncx = nx - nix;
+        if (nix > nx) {
+            pagmo_throw(std::invalid_argument, "The integer part cannot be larger than the bounds size");
+        }
+        for (decltype(ncx) i = ncx; i < nx; ++i) {
+            if (lb[i] != std::floor(lb[i])) {
+                pagmo_throw(std::invalid_argument,
+                            "A lower bound of the integer part of the decision vector is: " + std::to_string(lb[i])
+                                + " and is not an integer.");
+            }
+            if (ub[i] != std::floor(ub[i])) {
+                pagmo_throw(std::invalid_argument,
+                            "An upper bound of the integer part of the decision vector is: " + std::to_string(ub[i])
+                                + " and is not an integer.");
+            }
+        }
+    }
```

Comment:
In my mind we did not have to check for infinite values as we do allow infinte bounds for floats and our integers are just floats.  But it would also make sense to forbid infinite values in the case of integers .... breaking though the symmetry int / floats.